pkg/host/internal/bridge/bridge.go

```diff
@@ -0,0 +1,81 @@
+package bridge
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/bridge/ovs"
+	ovsStorePkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/bridge/ovs/store"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/types"
+)
+
+type bridge struct {
+	ovs ovs.Interface
+}
+
+// New return default implementation of the BridgeInterface
+func New() types.BridgeInterface {
+	return &bridge{
+		ovs: ovs.New(ovsStorePkg.New()),
+	}
+}
+
+// DiscoverBridges returns information about managed bridges on the host
+func (b *bridge) DiscoverBridges() (sriovnetworkv1.Bridges, error) {
+	log.Log.V(1).Info("DiscoverBridges(): discover managed bridges")
+	discoveredOVSBridges, err := b.ovs.GetOVSBridges(context.Background())
+	if err != nil {
+		log.Log.Error(err, "DiscoverBridges(): failed to discover managed OVS bridges")
+		return sriovnetworkv1.Bridges{}, err
+	}
+	return sriovnetworkv1.Bridges{OVS: discoveredOVSBridges}, nil
+}
+
+// ConfigureBridge configure managed bridges for the host
+func (b *bridge) ConfigureBridges(bridgesSpec sriovnetworkv1.Bridges, bridgesStatus sriovnetworkv1.Bridges) error {
+	log.Log.V(1).Info("ConfigureBridges(): configure bridges")
+	if len(bridgesSpec.OVS) == 0 && len(bridgesStatus.OVS) == 0 {
+		// there are no reported OVS bridges in the status and the spec doesn't contains bridges.
+		// no need to validated configuration
+		log.Log.V(2).Info("ConfigureBridges(): configuration is not required")
+		return nil
+	}
+	for _, curBr := range bridgesStatus.OVS {
+		found := false
+		for _, desiredBr := range bridgesSpec.OVS {
+			if curBr.Name == desiredBr.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			if err := b.ovs.RemoveOVSBridge(context.Background(), curBr.Name); err != nil {
+				log.Log.Error(err, "ConfigureBridges(): failed to remove OVS bridge", "bridge", curBr.Name)
+				return err
+			}
+		}
+	}
+	// create bridges, existing bridges will be updated only if the new config doesn't match current config
+	for _, desiredBr := range bridgesSpec.OVS {
+		desiredBr := desiredBr
```

Comment:
Unless we're going to run go routines, we don't need this new variable.
If `err := b.ovs.CreateOVSBridge(...)` spawns go routines, we can take care of the loop problem there. Or at least add a comment here to describe why we need this.

Reply:
yes, that's true. But as a caller we can't know how the pointer will be used in the function to which we passed it. I usually prefer to not pass pointer to a loop variable. I updated the code to avoid reassign of the var .